transformer/transforms/util/import_csv.py

```diff
@@ -0,0 +1,107 @@
+from transformer.registry import register
+from transformer.transforms.base import BaseTransform
+import csv
+import urllib
+import tempfile
+
+#Approximate size of a 1000 line CSV with 14 columns, some tezt in each
+#This takes ~3-5 seconds to turn into line-items in Zapier, and is still workable in the editor
+#So not arbitrary, but I'm sure it could be tweaked
+MAX_CSV_FILE_SIZE = 150000
+
+class UtilImportCSVTransform(BaseTransform):
+
+    category = "util"
+    name = "import_csv"
+    label = "Import CSV File"
+    help_text = (
+        "Import a CSV file from a public URL, File field from another Zap step, or entered text. "
+        "Limited to 150k/1000 rows. "
+        "More on using csv files [here] (https://zapier.com/help/formatter/#how-process-csvs-formatter)"
+    )
+
+    noun = "CSV"
+    verb = "import"
+
+    def build_input_field(self):
+        return {
+            "type": "file",
+            "required": True,
+            "key": "inputs",
+            "label": "CSV file",
+        }
+
+    def transform(self, csv_url, line_items, **kwargs):
+        # Take a file input and output a set of line-item fields, or a big string field
+
+        if not csv_url:
+            return u''
+
+        # create a temp file, then load the csv into it
+        url = csv_url
+        response = tempfile.NamedTemporaryFile()
+        response.seek(0)
+        urllib.urlretrieve(url, response.name)
+
+        #check file size
+        response.seek(0, 2)
+        size = response.tell()
+        if (size > MAX_CSV_FILE_SIZE):
+            self.raise_exception('CSV Import only supports file sizes < 150K.')
+
+        # use csv utils to see if there is a header, and get the dialect (format) of the csv
+        response.seek(0)
+        try:
+            header = csv.Sniffer().has_header(response.read())
+        except:
```

Comment:
Instead of catching _all_ exceptions, let's try to catch only the ones we're expecting. If we don't this could be difficult to debug since an unanticipated exception would lead us here. what was the error you got when you tried to read an unparsable CSV? Whatever that is we can put that here:
```suggestion
        except IOError:
```

Reply:
The csv library eats this error, and just replies:

{
    "message": "Could not determine delimiter"
}

So, I think rather than try to try/except, I'll just let this error bubble up. This is what is recommended here:
https://codereview.stackexchange.com/questions/172936/opening-a-csv-file-in-python-with-error-handling